[SRVCOM-1846]: Remove outdated module and add content to func getting started (#46032)
openshift/openshift-docs: modules/serverless-build-func-kn.adoc

[id="serverless-build-func-kn_{context}"]
= Building functions

> Before you can run a function, you must build the function project by using the `kn func build` command. This command creates an OCI container image that can be run locally on your computer, or on a Kubernetes cluster. The build command uses the function project name and the image registry name to construct a fully qualified image name for your function.

Reviewer: NBD but you don't need a comma before `or on a Kubernetes cluster.`

Author: Good catch, also updated Kubernetes -> {product-title} since these are product docs 😉

[id="serverless-build-func-kn_{context}"]
= Building functions

> Before you can run a function, you must build the function project by using the `kn func build` command. The build command reads the `func.yaml` file from the function project directory to determine the image name and the registry used to store functions images.

Reviewer 2: In the spirit of keeping func.yaml out of the examples as much as possible...
```suggestion
Before you can run a function, you must build the function project using the `kn func build` command. The build command uses the function project name and the image registry name to construct a fully qualified image name for your function.

.Example `func.yaml`
----
image: image-registry.openshift-image-registry.svc:5000/<oc_project>/<function_name>
----
```

Author: I'd remove "Example `func.yaml`" entirely then. I don't think just showing this one `image` line adds anything useful, really it just removes any context for users to see _how_ it fits into func.yaml.

.Example `func.yaml`
[source,yaml]
----
name: example-function
namespace: default
runtime: node
image: image-registry.openshift-image-registry.svc:5000/<oc_project>/<function_name>
imageDigest: ""
trigger: http
builder: default
builderMap:
  default:
envs: {}
----

> 

Reviewer 2: If you accept my above suggestion, then delete all of this.

> By default, the OpenShift Container Registry is used in `func.yaml` when you create a new project. The registry used to store function images can be overridden by using the `-r` flag or the `FUNC_REGISTRY` environment variable.

Reviewer 2: ```suggestion
By default, the OpenShift Container Registry is used as the image registry for storing function images. This can be overridden using the `-r` flag or the `FUNC_REGISTRY` environment variable.
```

.Example build command
[source,terminal]
----

> $ kn func build [-i <image> -p <path>]

Reviewer 2: `func build` now has a lot of flags. I wonder if it makes sense to show any here.  I would suggest removing all flags from the example, or having some text such as "To learn more about the build command options, use the help command `kn func help build`".

----

.Example build command using the optional registry flag
[source,terminal]
----

> $ kn func build [-i <image> -r <registry> -p <path>]

Reviewer 2: ```suggestion
$ kn func build -r quay.io/username
```
I would use a specific example that correlates with the output sample below.
